activerecord/test/cases/connection_adapters/schema_cache_test.rb

```diff
@@ -45,17 +72,25 @@ def test_dump_and_load
 
         @cache = Marshal.load(Marshal.dump(@cache))
 
-        assert_equal 11, @cache.columns("posts").size
-        assert_equal 11, @cache.columns_hash("posts").size
-        assert @cache.data_sources("posts")
-        assert_equal "id", @cache.primary_keys("posts")
+        assert_no_queries do
+          assert_equal 11, @cache.columns("posts").size
+          assert_equal 11, @cache.columns_hash("posts").size
+          assert @cache.data_sources("posts")
+          assert_equal "id", @cache.primary_keys("posts")
+        end
       end
 
       def test_table_methods_deprecation
         assert_deprecated { assert @cache.table_exists?("posts") }
         assert_deprecated { assert @cache.tables("posts") }
         assert_deprecated { @cache.clear_table_cache!("posts") }
       end
+
+      private
+
+      def schema_dump_path
+        'test/assets/schema_dump_50.yml'
```

Comment:
The name of this file should be `'test/assets/schema_dump_5_1.yml'`. I don't think it is going to be safe to backport this.